### PR TITLE
Fix "can't modify frozen string" errors when pods are integrated using the `branch` option (#10920)

### PR DESCRIPTION
**Pull request by Buju77**
this PR fixes https://github.com/CocoaPods/CocoaPods/issues/10920

### CODE REVIEW

File: lib/cocoapods-downloader/git.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def self.preprocess_options(options)
       #
       def self.commit_from_ls_remote(output, branch_name)
         return nil if branch_name.nil?
-        encoded_branch_name = branch_name.force_encoding(Encoding::ASCII_8BIT)
+        encoded_branch_name = branch_name.encode(Encoding::ASCII_8BIT)
```
Comment (dnkoutso) [inline lib/cocoapods-downloader/git.rb:55]: let me see what I can do because I think we do not want to change the underlying encoding of this string.

**Comment by dnkoutso** [comment]
Follow up to https://github.com/CocoaPods/cocoapods-downloader/pull/116 (for posterity).

@Buju77 can you also try this fix instead?

```ruby
encoded_branch_name = branch_name.dup.force_encoding(Encoding::ASCII_8BIT)
```
?

**Comment by dnkoutso** [comment]
Also lets point to `1-5-stable` branch since we will do a quick 1.5.1 release for this fix.

**Comment by dnkoutso** [comment]
I think this fix will suffice.

**Comment by dnkoutso** [comment]
@Buju77 lets make sure we point to `1-5-stable` branch and rebase this PR before we merge it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,8 +8,9 @@
 
 ##### Bug Fixes
 
-* None.  
-
+* Fix "can't modify frozen string" errors when pods are integrated using the `branch` option
```
Comment (dnkoutso) [inline CHANGELOG.md:11]: nit: please add two empty spaces at the end of this line for markdown formatting.

**Comment by Buju77** [comment]
> Follow up to #116 (for posterity).
> 
> @Buju77 can you also try this fix instead?
> 
> ```ruby
> encoded_branch_name = branch_name.dup.force_encoding(Encoding::ASCII_8BIT)
> ```
> 

yes, this works as well.

But I thought that `branch_name.encode()` would already return a copy and doesn't modify `branch_name` itself?

**Comment by Buju77** [comment]
> @Buju77 lets make sure we point to `1-5-stable` branch and rebase this PR before we merge it.

ok, I've now rebased on top of `1-5-stable`

**Comment by dnkoutso** [comment]
@Buju77 I think you are right and we can fix it your way

**Review by dnkoutso — APPROVED** [review]

**Comment by dnkoutso** [comment]
oh well we can land this now! :D

**Comment by dnkoutso** [comment]
I will ship 1.5.1 of cocoapods-downloader

**Comment by dnkoutso** [comment]
@Buju77 shipped! https://github.com/CocoaPods/cocoapods-downloader/releases/1.5.1

**Comment by Buju77** [comment]
@dnkoutso thank you very much! 👍

**Pull request merged**